### PR TITLE
WIP: Add missing requirements.txt file

### PR DESCRIPTION
**Pull request by jamietanna**
In order to more easily track the required dependencies to get the
project up and running, use a `requirements.txt` file.

*Note*: This PR is currently a WIP as it would be good to hear thoughts on the best way to lock down dependencies for `pyalpm`, `sqlite` and `expac`, if possible.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,5 @@
 __pycache__/
 app.db
+
+virtualenv
+venv
```
Comment (anthraxx) [inline .gitignore]: guess just one of those is needed, i would propose to go with the hidden file ".virtualenv" instead

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+Flask-SQLAlchemy==2.1
+Flask-WTF==0.13.1
+itsdangerous==0.24
+Jinja2==2.8
+MarkupSafe==0.23
+requests==2.12.4
+scrypt==0.8.0
+SQLAlchemy==1.1.4
+Werkzeug==0.11.11
+WTForms==2.1
```
Comment (anthraxx) [inline requirements.txt]: - please don't use frozen dependency versions, its meant to work with the latest. If not, its a bug.
- please only use direct first-level dependencies and purge the transitive dependencies like click, MarkupSafe etc...

File: README.md [hunk 3]
```diff
@@ -27,6 +27,14 @@ vulnerability details and generating security advisories.
 * sqlite
 * expac
 
+Python dependencies can be installed in a virtual environment (`virtualenv`), by running:
+
+```python
+virtualenv venv
+. venv/bin/activate
+pip install -r requirements.txt
+```
```
Comment (anthraxx) [inline README.md]: actually this is not a python code block, this should just use the regular un-syntaxed code block

File: .gitignore (hunk 1 above)
Comment (jamietanna) [inline .gitignore]: Should be sorted as of latest commits.

**Comment by anthraxx** [comment]
Looks good, but can we get scrypt back? :P

**Comment by jamietanna** [comment]
Woops, missed that! Should be all there now, @anthraxx.

**Comment by anthraxx** [comment]
so all looks fine now and the ```requests.txt``` is certainly something we could add, however i'm not yet convinced about the virtualenv instructions as without pyalpm this will simply fail to be used. we need proper instruction for pyalpm... hm...

**Comment by jamietanna** [comment]
Agreed - is there any way we can get `pyalpm` distributed on PyPi? That would mean we can pull it in as a dependency more easily.

**Comment by anthraxx** [comment]
let me see if i can make that happen, need to discuss that

**Comment by anthraxx** [comment]
well I would be willing to accept a temporary entry for ```https://git.archlinux.org/users/remy/pyalpm.git/snapshot/pyalpm-0.8.tar.xz``` so that the virtualenv instruction works.
We could replace that URL later with a simple name if its available on pypi by its maintainer (already triggered to do so).

**Comment by jamietanna** [comment]
Sure, I'll push up a commit that adds the URL to `pyalpm`, and raise a second, `WIP` PR that we can track progress to getting `pyalpm` on PyPi. Sound good?

**Comment by anthraxx** [comment]
@jamietanna sure, just add that url for the requests.txt... I don't thin we need a meta-PR, I will notice it when we have pyalpm on pypi and i will definitivly not forget to kick the URL out as i hate it there :stuck_out_tongue_closed_eyes:

**Comment by jamietanna** [comment]
Your call, @anthraxx :) Think that's ready, unless you can see anything else?

**Comment by anthraxx** [comment]
applied via CLI, thanks a lot

**Pull request closed without merging**